nil pointer panic when file.content resource errors

``` hcl
param "message" {
  default = "Hello, World"
}

param "filename" {
  default = "/tmp/some-dir/test.txt"
}

file.content "render" {
  destination = "{{param `filename`}}"
  content     = "{{param `message`}}"
}
```

assuming `/tmp/some-dir` does not exist, you will get the following when running apply:

``` shell
$ ./runconv apply                                                                           [9/2793]
make: `converge' is up to date.
2016/08/19 12:52:51 [INFO] applying samples/fileContent.hcl
2016/08/19 12:52:51 [INFO] resolving dependencies
2016/08/19 12:52:51 [INFO] loading resources
2016/08/19 12:52:51 [INFO] rendering
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x80346]

goroutine 1 [running]:
panic(0x481500, 0xc4200140d0)
        /usr/local/Cellar/go/1.7/libexec/src/runtime/panic.go:500 +0x1a1
text/template.errRecover(0xc420107320)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:140 +0x2ad
panic(0x481500, 0xc4200140d0)
        /usr/local/Cellar/go/1.7/libexec/src/runtime/panic.go:458 +0x243
github.com/asteris-llc/converge/apply.(*Result).Messages(0xc4201cd2f0, 0x10f28, 0x20, 0xc42018bfb0)
        /Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/apply/result.go:32 +0x26
github.com/asteris-llc/converge/prettyprinters/human.(*printerNode).Messages(0xc42018f3e0, 0x0, 0x0, 0x0)
        <autogenerated>:8 +0x4a
reflect.Value.call(0x499100, 0xc42018f3e0, 0xe13, 0x5022bf, 0x4, 0x776e00, 0x0, 0x0, 0xf2693, 0x75a180, ...)
        /usr/local/Cellar/go/1.7/libexec/src/reflect/value.go:434 +0x5c8
reflect.Value.Call(0x499100, 0xc42018f3e0, 0xe13, 0x776e00, 0x0, 0x0, 0xc420196078, 0x4, 0x4)
        /usr/local/Cellar/go/1.7/libexec/src/reflect/value.go:302 +0xa4
text/template.(*state).evalCall(0xc4201072a0, 0x499100, 0xc42018f3e0, 0x16, 0x499100, 0xc42018f3e0, 0xe13, 0x738880, 0xc4201cdbf0, 0x51ffc4, ...)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:658 +0x530
text/template.(*state).evalField(0xc4201072a0, 0x499100, 0xc42018f3e0, 0x16, 0x51ffc4, 0x8, 0x738880, 0xc4201cdbf0, 0xc42019b2d0, 0x1, ...)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:549 +0xdb6
text/template.(*state).evalFieldChain(0xc4201072a0, 0x499100, 0xc42018f3e0, 0x16, 0x499100, 0xc42018f3e0, 0x16, 0x738880, 0xc4201cdbf0, 0xc42019b2c0, ...)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:520 +0x22f
text/template.(*state).evalFieldNode(0xc4201072a0, 0x499100, 0xc42018f3e0, 0x16, 0xc4201cdbf0, 0xc42019b2d0, 0x1, 0x1, 0x0, 0x0, ...)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:484 +0x114
text/template.(*state).evalCommand(0xc4201072a0, 0x499100, 0xc42018f3e0, 0x16, 0xc4201cdbc0, 0x0, 0x0, 0x0, 0x4c3540, 0xc4201cd2c0, ...)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:422 +0x866
text/template.(*state).evalPipeline(0xc4201072a0, 0x499100, 0xc42018f3e0, 0x16, 0xc4201ce690, 0xc42018f3e0, 0x16, 0x29cdc)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:400 +0xee
text/template.(*state).walkRange(0xc4201072a0, 0x499100, 0xc42018f3e0, 0x16, 0xc4201a9940)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:309 +0xf3
text/template.(*state).walk(0xc4201072a0, 0x499100, 0xc42018f3e0, 0x16, 0x738b20, 0xc4201a9940)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:237 +0x59c
text/template.(*state).walk(0xc4201072a0, 0x499100, 0xc42018f3e0, 0x16, 0x7389a0, 0xc4201cd560)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:234 +0x13d
text/template.(*Template).execute(0xc4201a9740, 0x730ba0, 0xc420070070, 0x499100, 0xc42018f3e0, 0x0, 0x0)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:189 +0x204
text/template.(*Template).Execute(0xc4201a9740, 0x730ba0, 0xc420070070, 0x499100, 0xc42018f3e0, 0x0, 0xc4201073a0)
        /usr/local/Cellar/go/1.7/libexec/src/text/template/exec.go:175 +0x53
github.com/asteris-llc/converge/prettyprinters/human.(*Printer).DrawNode(0xc42017e800, 0xc4201ccf00, 0xc420164c60, 0x18, 0xea0108, 0xc4201d2220, 0x0, 0x0)
        /Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/prettyprinters/human/human.go:118 +0x2ee
github.com/asteris-llc/converge/prettyprinters.Printer.Show(0x4c1cc0, 0xc42017e800, 0x7376a0, 0xc420077040, 0xc4201ccf00, 0xc42008a080, 0x0, 0x546714, 0x74cad8)
        /Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/prettyprinters/prettyprinter.go:54 +0x375
github.com/asteris-llc/converge/cmd.glob..func2(0x753440, 0xc4200f77e0, 0x1, 0x2)
        /Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/cmd/apply.go:95 +0x5f1
github.com/asteris-llc/converge/vendor/github.com/spf13/cobra.(*Command).execute(0x753440, 0xc4200f7560, 0x2, 0x2, 0x753440, 0xc4200f7560)
        /Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/vendor/github.com/spf13/cobra/command.go:565 +0x411
github.com/asteris-llc/converge/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x753c40, 0xc420074058, 0x0, 0xc420107f18)
        /Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/vendor/github.com/spf13/cobra/command.go:651 +0x367
github.com/asteris-llc/converge/vendor/github.com/spf13/cobra.(*Command).Execute(0x753c40, 0x0, 0x0)
        /Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/vendor/github.com/spf13/cobra/command.go:610 +0x2b
github.com/asteris-llc/converge/cmd.Execute()
        /Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/cmd/root.go:59 +0x31
main.main()
        /Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/main.go:20 +0x14
```

--- a/apply/result.go
+++ b/apply/result.go
@@ -29,10 +29,20 @@ type Result struct {
 }
 
 // Messages returns any result status messages supplied by the task
-func (r *Result) Messages() []string { return r.Status.Messages() }
+func (r *Result) Messages() []string {
+	if r.Status != nil {
+		return r.Status.Messages()
+	}
+	return nil
+}
 
 // Changes returns the fields that changed
-func (r *Result) Changes() map[string]resource.Diff { return r.Plan.Changes() }
+func (r *Result) Changes() map[string]resource.Diff {
+	if r.Plan != nil {
+		return r.Plan.Changes()
+	}
+	return nil
+}
 
 // HasChanges indicates if this result ran
 func (r *Result) HasChanges() bool { return r.Ran }

